aspnetcore/blazor/fundamentals/signalr.md

```diff
@@ -1127,6 +1127,31 @@ Blazor Server:
 
 **In the preceding example, the `{BLAZOR SCRIPT}` placeholder is the Blazor script path and file name.** For the location of the script and the path to use, see <xref:blazor/project-structure#location-of-the-blazor-script>.
 
+:::moniker range=">= aspnetcore-9.0"
+
+When the user navigates back to an app with a disconnected circuit, reconnection is attempted immediately rather than waiting for the duration of the next reconnect interval. This improves the user experience when navigating to an app in a browser tab that has gone to sleep.
+
+When a reconnection attempt reaches the server but the server has already released the circuit, a page refresh occurs automatically. This prevents the user from having to manually refresh the page if it's likely going to result in a successful reconnection.
+
+Reconnect timing uses a computed backoff strategy. The first several reconnection attempts occur in rapid succession without a retry interval before computed delays are introduced between attempts. The default logic for computing the retry interval is an implementation detail subject to change without notice, but you can find the default logic that the Blazor framework uses [in the `computeDefaultRetryInterval` function (reference source)](https://github.com/search?q=repo%3Adotnet%2Faspnetcore%20computeDefaultRetryInterval&type=code).
+
+[!INCLUDE[](~/includes/aspnetcore-repo-ref-source-links.md)]
+
+Customize the retry interval behavior by specifying a function to compute the retry interval. In the following exponential backoff example, the number of previous reconnection attempts is multiplied by 1,000 ms to calculate the retry interval. When the count of previous attempts to reconnect (`previousAttempts`) is greater than the maximum retry limit (`maxRetries`), `null` is assigned to the retry interval (`retryIntervalMilliseconds`) to cease further reconnection attempts:
+
+```javascript
+Blazor.start({
+  circuit: {
+    reconnectionOptions: {
+      retryIntervalMilliseconds: (previousAttempts, maxRetries) => 
+        previousAttempts >= maxRetries ? null : previousAttempts * 1000
+    },
+  },
+});
```

Comment:
Totally optional, but we could also mention another neat little trick if you want more precise control over each retry interval:

```js
Blazor.start({
  circuit: {
    reconnectionOptions: {
      retryIntervalMilliseconds: Array.prototype.at.bind([0, 1000, 2000, 5000, 10000, 15000, 30000]),
    },
  },
});
```

This just lets you just list out the retry intervals directly. After the last specified retry interval occurs, retries stop (because the function returns `undefined`).

Reply:
Yes, I'll add that. That's nice.